### ISSUE TITLE
enable configuration cache

### ISSUE THREAD

**Issue by martinbonnin** [apollographql/apollo-kotlin#4416]
Blocked on https://youtrack.jetbrains.com/issue/KT-55051/MetadataDependencyTransformationTask-is-not-compatible-with-Gradle-configuration-cache

**Comment by BoD** [comment]
Done in #5083

**Comment by BoD** [comment]
Actually there are still a few issues left that prompted us to disable it for now:
- https://github.com/Kotlin/dokka/issues/1217
- https://youtrack.jetbrains.com/issue/KT-60311/

See also https://github.com/gradle/gradle/issues/20673.

See #5089

**Comment by BoD** [comment]
Done again in #5530

**Comment by dsvensson** [comment]
Seems to be a bit more to do here looking forward. Enabling upcoming configuration cache related project isolation via `org.gradle.unsafe.isolated-projects=true` fails on apollo gradle plugin. Perhaps should be a new issue.

More details, https://docs.gradle.org/nightly/userguide/isolated_projects.html

**Comment by martinbonnin** [comment]
@dsvensson project isolation compatibility is tracked [there](https://github.com/apollographql/apollo-kotlin/issues/6180). Can you add the details of the error (stacktrace, build report) there?
